src/tests/ftest/checksum/csum_error_logging.py

```diff
@@ -62,17 +62,17 @@ def test_csum_error_logging(self):
         host_devices = get_dmg_smd_info(dmg.storage_query_list_devices, 'devices')
         for host, devices in host_devices.items():
             for device in devices:
-                for entry in ('uuid', 'tgt_ids', 'role_bits', 'roles'):
+                for entry in ('uuid', 'tgt_ids', 'role_bits'):
                     if entry not in device:
                         self.fail(
                             'Missing {} info from dmg storage query list devices'.format(entry))
                 self.log.info(
-                    'Host %s device: uuid=%s, targets=%s, role=%s, role_bits=%s',
-                    host, device['uuid'], device['tgt_ids'], device['roles'], device['role_bits'])
+                    'Host %s device: uuid=%s, targets=%s, role_bits=%s',
+                    host, device['uuid'], device['tgt_ids'], device['role_bits'])
                 if not device['tgt_ids']:
                     self.log_step('Skipping device without targets on {}'.format(device['uuid']))
                     continue
-                if device['roles'] and not int(device['role_bits']) & 1:
+                if not int(device['role_bits']) & 1:
```

Comment:
The problem in the ticket is `roles` used to be assigned to `''` when unset and now it's `NA`, so we only need to support treating `NA` the same as when `device['roles']` is not set:
```suggestion
                if (device['roles'] != 'NA') and not int(device['role_bits']) & 1:
```

Reply:
This change I don't feel why we need to do? Yes, it was NULL before and now it is NA... What is the point in checking it? At some point if the value changes, the script will once again will be broken. Also, if we want to re-test with previous version of DAOS, the test will fail. I am not clear about the reasoning behind this check. I can add the check but I don't see the use.

Reply:
I think the correct solution is to keep your original changes and just adjust the check to `if int(device['role_bits']) > 0 and not (int(device['role_bits']) & 1):` then you are not relying on any changes in syntax of `roles` and instead just relying on the behaviour of `role_bits` which is less likely to change.

Reply:
In MD-on-PMem mode, role_bits will always be 0. Roles are only assigned in MD-on-SSD mode so given that roles == "NA" whenever role_bits = 0 you can just test if `bits > 0 && !(bits & 1)`.

Reply:
Thanks @tanabarr ... I think this seems to be the correct suggestion. I will proceed with this change.